_episodes/40.md

```diff
@@ -0,0 +1,481 @@
+
+* The improved performance of inserts on Delta Lake, Hive, and Iceberg is a huge
+one. We're not entirely sure how much it'll matter in production use cases, but
+some of the benchmarks suggested it could be massive - one test showed a 75%
+reduction in query duration.
+* Dynamic function resolution in the SPI is going to unlock some very neat
+possibilities down the line. 
+
+More detailed information is available in the release notes for
+[Trino 394]({{site.url}}/docs/current/release/release-394.html),
+and
+[Trino 395]({{site.url}}/docs/current/release/release-395.html).
+
+## Concept of the week: Latest features in Apache Iceberg and the Iceberg connector
+
+It has been over a year since we had Ryan on the Trino Community Broadcast as
+guest to discuss what Apache Iceberg is, and how it can be used in Trino. Since
+then, the adoption of Iceberg in our community has skyrocketed. Iceberg is truly
+delivering as a much better alternative to the Hive table format. 
+
+The initial phase of the Iceberg connector in Trino aimed to provide fast and 
+interoperable read support. A typical usage was Trino alongside other query 
+engines like Apache Spark which supported many of the data modification language
+(DML) SQL features on Iceberg. One of the biggest requests we got as adoption
+increased was the ability to do everything through Trino. This episode dives
+into some of the latest features that were missing from the early iterations of
+the Iceberg connector and what has changed in Iceberg as well!
+
+### What is Apache Iceberg?
+
+[Iceberg](https://iceberg.apache.org/) is a next-generation table format that 
+defines a standard around the metadata used to map data to a SQL query engine. 
+It addresses a lot of the maintainability and reliability issues many engineers
+experienced with the way
+[Hive modeled SQL tables]({% post_url 2020-10-20-intro-to-hive-connector %}) 
+over big data files. 
+
+One common confusion to point out is that table format is not equivalent to file
+formats like ORC or Parquet. The table format is the layer that maintains 
+metadata mapping these files to the concept of a table and other common database
+abstractions.
+
+This episode assumes you have some basic knowledge of Trino and Iceberg already. If
+you are new to Iceberg or need a refresher, we recommend the two older episodes
+about Iceberg and Trino basics:
+
+* [14: Iceberg: March of the Trinos]({{site.url}}/episodes/14.html)
+* [15: Iceberg right ahead!]({{site.url}}/episodes/15.html)
+
+### Why Iceberg over other formats?
+
+There has been some great advancements to big data technologies that brought
+back some SQL and data warehouse capabilities. However, Hive and Hive-like table
+formats are still missing some capabilities due to limitations that Hive tables
+have, such as dropping and reintroducing stale data unintentionally. On top of
+that, Hive tables require a lot of knowledge of Hive internals. More recent
```

Comment:
```suggestion
that, Hive tables require a lot of knowledge of Hive internals. Some recent
```

Reply:
Somehow we need to add that Iceberg does not fall into that category .. maybe to start the next paragraph?